### ISSUE TITLE
BlockPreview should render styles

### ISSUE THREAD

**Issue by javierarce** [WordPress/gutenberg#30680]
## Description

While reviewing [this issue](https://github.com/WordPress/gutenberg/issues/30108) we noticed that the previews in the Full Site Editor (FSE) template sidebar don't truly represent the design in the editor. 

The reason seems to be that the [BlockPreview](https://github.com/WordPress/gutenberg/tree/2e6676a342344e27515b7c8c953755b353fd693b/packages/block-editor/src/components/block-preview) component doesn't load styles (in this case, the ones from the editor) or the way this component is used in the [template preview](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/navigation-sidebar/navigation-panel/template-preview.js). In any case, fixing the previews by loading the editor styles could help users better identify the different templates while they work with the FSE.

## Step-by-step reproduction instructions

1. Set up a test site with the latest version of Gutenberg
2. Install and activate a theme that supports FSE (like the TT1 blocks theme).
3. Head to the Site Editor (beta) tab and open any template.
4. Change the Global Styles and set the background color to red.
5. Click on the WordPress menu to open the template sidebar.
6. Hover some templates to make the preview appear.

## Expected behaviour

The template previews should show a red background.

## Actual behaviour

The template previews show a white background. 

## Screenshots or screen recording

![Screenshot 2021-04-09 at 17 59 35](https://user-images.githubusercontent.com/4933/114208063-5ad74e00-995d-11eb-8364-53a00bf4ef65.png)

The code rendered by the component doesn't seem to include any CSS related to the theme:

![Screenshot 2021-04-09 at 18 15 11](https://user-images.githubusercontent.com/4933/114210036-9a069e80-995f-11eb-92da-843e409bdbfe.png)

## WordPress information

- WordPress version: 5.7
- Gutenberg version: 10.3.2 
- Are all plugins except Gutenberg deactivated? Yes
- Are you using a default theme (e.g. Twenty Twenty-One)? No (see step-by-step instructions)

CC @annezazu @jasmussen

**Comment by jasmussen** [comment]
Thanks for ticketing this! 

On the one hand this seems like a logical extension of the stylesheet being loaded into the site editor iframe, and so the `editor-styles-wrapper` classes of the thumbnail component don't pick up those sandboxed styles. 

On the other hand, I could _swear_ this used to be working, not that long ago. @ellatrix do you recall if this broke with the iframe and have been broken since?

**Comment by ramonjd** [comment]
I can replicate this for most patterns where the theme adds specific styling. I haven't confirmed, but I suspect that the theme css isn't being loaded in the iframe context?

This is tested with the current master branch (10.6.0-rc.1) using TT1 blocks, and no modifications:

| Site editor        | Block editor           | 
| ------------- |:-------------:|
|<img width="353" alt="Screen Shot 2021-05-11 at 1 24 31 pm" src="https://user-images.githubusercontent.com/6458278/117755042-db3fe600-b25e-11eb-9e8a-027a71d76259.png"> | <img width="382" alt="Screen Shot 2021-05-11 at 12 23 10 pm" src="https://user-images.githubusercontent.com/6458278/117755032-d5e29b80-b25e-11eb-96ea-3b7d570ec72a.png"> | 
| <img width="349" alt="Screen Shot 2021-05-11 at 1 45 25 pm" src="https://user-images.githubusercontent.com/6458278/117755502-ac763f80-b25f-11eb-9508-a3b65e7e17e9.png">     | <img width="315" alt="Screen Shot 2021-05-11 at 1 45 51 pm" src="https://user-images.githubusercontent.com/6458278/117755509-aed89980-b25f-11eb-93d1-51f2bdb41283.png">      | 

> On the one hand this seems like a logical extension of the stylesheet being loaded into the site editor iframe, and so the editor-styles-wrapper classes of the thumbnail component don't pick up those sandboxed styles.

There's a related issue over here: https://github.com/WordPress/gutenberg/issues/31022

**Comment by jasmussen** [comment]
> I can replicate this for most patterns where the theme adds specific styling. I haven't confirmed, but I suspect that the theme css isn't being loaded in the iframe context?

Well the popover is not part of the iframe. So even if the editor style is loaded inside the iframe, it's not loaded outside the iframe:

<img width="2404" alt="Screenshot 2021-05-11 at 09 17 30" src="https://user-images.githubusercontent.com/1204802/117774424-c6049080-b239-11eb-9626-bfc1bbed5725.png">

So it seems the solution is to either also load the editor style outside the iframe, in the site editor, or wrap each preview inside other iframes that load the editor styles.

**Comment by ramonjd** [comment]
> Well the popover is not part of the iframe

Ah sorry for the red herring there. Thanks for the extra info.

**Comment by ellatrix** [comment]
Should be fixed after #28165.
